### PR TITLE
[ZEPPELIN-5307] Ignore the single quote and double quote in sql comment

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?

This PR is to fix the bug of SqlSplitter when single/double quote in sql comment. UT is added. 

### What type of PR is it?
[Bug Fix]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-5307

### How should this be tested?
* CI pass

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/util/SqlSplitter.java [hunk 1]
```diff
@@ -102,15 +102,15 @@ public SqlSplitter(String... additionalSingleCommentPrefixList) {
         multiLineComment = false;
       }
 
-      if (character == '\'') {
+      if (character == '\'' && (!singleLineComment && !multiLineComment)) {
```
Comment (cuspymd) [inline zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/util/SqlSplitter.java:105]: As shown below, moving "!" out of parentheses will make it easier to understand the conditions.
```
if (character == '\'' && !(singleLineComment || multiLineComment)) {
```

Comment (zjffdu) [reply zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/util/SqlSplitter.java]: Addressed

**Review by cuspymd — APPROVED** [review]
Looks good!!

**Comment by zjffdu** [comment]
Thanks for the review @cuspymd Will merge if no more comment

**Pull request closed without merging**